### DISCUSSION TITLE
IOperationInvoker and DI

### DISCUSSION THREAD

**Discussion by oskarjf** [CoreWCF/CoreWCF#1225] [Q&A] [answered]
I'm upgrading a WCF project to CoreWCF. In the current project there is a custom implementation of IOperationInvoker.
This class has a baseInvoker of the type IOperationInvoker as constructor parameter and that is causing a circular dependency.  Do you have any samples of how it is best to set up a custom operation invoker?
I'm using Microsoft DI.

Regards, Oskar

**Comment by g7ed6e** [comment, marked as answer]
@oskarjf I don't think `IOperationInvoker` are pulled from DI. Please have a look to these UT to wrap the `IOperationInvoker` provided by CoreWCF with yours https://github.com/CoreWCF/CoreWCF/blob/ac5453e5520db73132e2739d536078e64c90ded1/src/CoreWCF.Primitives/tests/Extensibility/OperationInvokerTests.cs

**Reply by oskarjf** [reply]
Yes, I ended up with just newing the operation invoker with the required dependencies
